linux/deploy_vm/ubuntu/prepare_ubuntu_iso_install.yml

```diff
@@ -14,19 +14,21 @@
       {%- else -%}unknown
       {%- endif -%}
 
-- name: "Set Ubuntu autoinstall method"
-  ansible.builtin.set_fact:
-    ubuntu_install_method: |-
-      {%- if ubuntu_version and ubuntu_version is version('20.04', '>=') -%}cloud-init
-      {%- elif ubuntu_version and ubuntu_version is version('20.04', '<') -%}simulation
-      {%- else -%}cloud-init
-      {%- endif -%}
-  when: ubuntu_edition == 'server'
+- name: "Set fact of Ubuntu Server"
+  block:
+    - name: "Set Ubuntu autoinstall method"
+      ansible.builtin.set_fact:
+        ubuntu_install_method: |-
+          {%- if ubuntu_version and ubuntu_version is version('20.04', '>=') -%}cloud-init
+          {%- elif ubuntu_version and ubuntu_version is version('20.04', '<') -%}simulation
+          {%- else -%}cloud-init
```

Comment:
Since the default server install method is cloud-init, you can leave one if here for ubuntu version < 20.04, else the install method is `cloud-init`.

Reply:
Thanks. Done